### PR TITLE
refactor: Upload sourcegraph LSIF index

### PR DESCRIPTION
**Pull request by tanishiking**
This PR unlocks precise code intelligence on Sourcegraph, you can check the LSIF upload status here:
https://sourcegraph.com/github.com/scalameta/metals-vscode/-/code-intelligence/uploads
(sourcegraph boosts my code-review productivity a lot 🚀)

see: https://about.sourcegraph.com/blog/announcing-scip-typescript

---

Uploaded LSIF against `sourcegraph` branch on `tanishiking/metals-vscode`, and you can try precise code-intelligence on
https://sourcegraph.com/github.com/tanishiking/metals-vscode@sourcegraph Looks like it works great :tada:
Screen Shot 2022-07-12 at 14 26 17

### CODE REVIEW

File: .github/workflows/sourcegraph.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          node-version: "14"
+      - run: yarn install
+      - name: index
+        run: yarn run scip-typescript index
+      - name: src lsif upload
+        run: |
+          mkdir -p bin
+          curl -L https://sourcegraph.com/.api/src-cli/src_linux_amd64 -o bin/src
+          chmod +x bin/src
+          ./bin/src lsif upload -trace=3 -ignore-upload-failure -github-token $GITHUB_TOKEN
```
Comment (tanishiking) [inline .github/workflows/sourcegraph.yml:25]: Copied from `scalameta/metals`

**Review by kpodsiad — APPROVED** [review]
LGTM

**Comment by kpodsiad** [comment]
> (sourcegraph boosts my code-review productivity a lot 🚀)

BTW, how do you use source for code reviews? Currently, I'm using Github extension for vscode which allows me to easily checkout to PR branch and then view changes as well as diff. Curious how your workflow look like!

**Comment by tanishiking** [comment]
I use sourcegraph browser extension https://docs.sourcegraph.com/integration/browser_extension
This extension allows me to use basic code navigayion features on diff view of Github.
Before I use Sourcegraph extension, I usually fetch the remote branch to local amd launch vscode to enjoy code navigations on changes (so I can understand the change more detail), but now I don't need to checkout the remote branch to local :rocket:

**Pull request merged**